zk/legacy_executor_verifier/executor.go

```diff
@@ -46,21 +47,22 @@ type Executor struct {
 	conn       *grpc.ClientConn
 	connCancel context.CancelFunc
 	client     executor.ExecutorServiceClient
+	semaphore  chan struct{}
 }
 
 func NewExecutors(cfg Config) []*Executor {
 	executors := make([]*Executor, len(cfg.GrpcUrls))
 	var err error
 	for i, grpcUrl := range cfg.GrpcUrls {
-		executors[i], err = NewExecutor(grpcUrl, cfg.Timeout)
+		executors[i], err = NewExecutor(grpcUrl, cfg.Timeout, cfg.MaxConcurrentRequests)
```

Comment:
The code allows executors[i] to be nil, which will to exception in the following functions. Currently this is not an issue because "NewExecutor" could never return "nil" executor. Also NewExecutor could never return an error. Won't it be better to remove the error return from the NewExecutor?